### PR TITLE
CA-289145: Close socket if error occurs when connecting

### PR DESCRIPTION
**Pull request by krizex**
Signed-off-by: Yang Qian <yang.qian@citrix.com>

### CODE REVIEW

File: ocaml/lib/channel.ml [hunk 1]
```diff
@@ -33,8 +33,14 @@ let t_of_rpc x =
...
       let s = Unix.socket Unix.PF_INET Unix.SOCK_STREAM 0 in
-      Unix.connect s (Unix.ADDR_INET(Unix.inet_addr_of_string ip, port));
-      s
+      begin
+        try
+          Unix.connect s (Unix.ADDR_INET(Unix.inet_addr_of_string ip, port));
+          s
+        with e ->
+          Unix.close s;
+          raise e
```
Comment (mseri) [inline ocaml/lib/channel.ml]: We should probably wrap this into an `Internal_error` and raise the internal error instead

Comment (krizex) [reply ocaml/lib/channel.ml]: yes

**Comment by mseri** [comment]
@gaborigloi remember to pick this change if needed in the new xapi-storage

**Comment by mseri** [comment]
travis failure on `gen_html`, maybe due to the new version of `Cow`? Somebody please fix it
```
jbuilder build @gen_html
      ocamlc generator/html/.www.eobjs/www.{cmi,cmo,cmt} (exit 2)
(cd _build/default && /home/travis/.opam/4.04.2/bin/ocamlc.opt -w -40 -g -bin-annot -I generator/html/.www.eobjs -I /home/travis/.opam/4.04.2/lib/bytes -I /home/travis/.opam/4.04.2/lib/caml2html -I /home/travis/.opam/4.04.2/lib/cow -I /home/travis/.opam/4.04.2/lib/cstruct -I /home/travis/.opam/4.04.2/lib/ezjsonm -I /home/travis/.opam/4.04.2/lib/hex -I /home/travis/.opam/4.04.2/lib/jsonm -I /home/travis/.opam/4.04.2/lib/omd -I /home/travis/.opam/4.04.2/lib/parsexp -I /home/travis/.opam/4.04.2/lib/ppx_sexp_conv/runtime-lib -I /home/travis/.opam/4.04.2/lib/re -I /home/travis/.opam/4.04.2/lib/re/posix -I /home/travis/.opam/4.04.2/lib/sexplib -I /home/travis/.opam/4.04.2/lib/sexplib0 -I /home/travis/.opam/4.04.2/lib/stringext -I /home/travis/.opam/4.04.2/lib/uchar -I /home/travis/.opam/4.04.2/lib/uri -I /home/travis/.opam/4.04.2/lib/uutf -I /home/travis/.opam/4.04.2/lib/xmlm -I generator/lib/.generator_lib.objs -no-alias-deps -o generator/html/.www.eobjs/www.cmo -c -impl generator/html/www.ml)
File "generator/html/www.ml", line 327, characters 12-14:
Error: This expression has type ('a Cow__.Xml.frag as 'a) Cow__.Xml.frag list
       This is not a function; it cannot be applied.
```

**Comment by gaborigloi** [comment]
We'll remove this when the sxm branch gets merged, so I think it's fine to ignore

**Pull request merged**